numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -73,6 +73,7 @@ NPY_NO_EXPORT int NPY_NUMUSERTYPES = 0;
  */
 #include "funcs.inc"
 #include "loops.h"
+#include "matmul.h"
```

Comment:
Does this have to be included? It would seem it is not used at all.

Reply:
It is needed for the `matmul` inner loop expansions, just like `loops.h` is needed for the other inner loop functions definitions.

Reply:
I'm still confused and am probably just missing something trivial, but `matmul.h.src` defines only items that have `matmul` in their name and I cannot find any occurrence of `matmul` in this file other than this include statement.

Reply:
Pushed the `loops.h` and `matmul.h` include into `__umath_generated.c` by adding them to `core/code_generators/generate_umath.py`